frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/traverse/TraverseSyncToRemote.java

```diff
@@ -0,0 +1,224 @@
+import static com.virtuslab.gitmachete.frontend.resourcebundles.GitMacheteBundle.getNonHtmlString;
+import static com.virtuslab.gitmachete.frontend.resourcebundles.GitMacheteBundle.getString;
+
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.ui.MessageConstants;
+import com.intellij.openapi.ui.MessageDialogBuilder;
+import com.intellij.openapi.vcs.VcsNotifier;
+import git4idea.GitLocalBranch;
+import git4idea.push.GitPushSource;
+import git4idea.repo.GitRepository;
+import lombok.CustomLog;
+import lombok.experimental.ExtensionMethod;
+import lombok.val;
+import org.checkerframework.checker.guieffect.qual.UIEffect;
+
+import com.virtuslab.gitmachete.backend.api.IBranchReference;
+import com.virtuslab.gitmachete.backend.api.IManagedBranchSnapshot;
+import com.virtuslab.gitmachete.backend.api.SyncToRemoteStatus;
+import com.virtuslab.gitmachete.frontend.actions.backgroundables.ResetCurrentToRemoteBackgroundable;
+import com.virtuslab.gitmachete.frontend.actions.common.FastForwardMerge;
+import com.virtuslab.gitmachete.frontend.actions.common.FetchUpToDateTimeoutStatus;
+import com.virtuslab.gitmachete.frontend.actions.common.MergeProps;
+import com.virtuslab.gitmachete.frontend.actions.dialogs.DivergedFromRemoteDialog;
+import com.virtuslab.gitmachete.frontend.actions.dialogs.GitPushDialog;
+import com.virtuslab.gitmachete.frontend.resourcebundles.GitMacheteBundle;
+import com.virtuslab.gitmachete.frontend.ui.api.table.BaseEnhancedGraphTable;
+
+@ExtensionMethod(GitMacheteBundle.class)
+@CustomLog
+public class TraverseSyncToRemote {
+
+  private final Project project;
+  private final GitRepository gitRepository;
+  private final BaseEnhancedGraphTable graphTable;
+  private final IBranchReference gitMacheteBranchOld;
+  private final Runnable traverseNextEntry;
+
+  public TraverseSyncToRemote(GitRepository gitRepository,
+      BaseEnhancedGraphTable graphTable,
+      IManagedBranchSnapshot gitMacheteBranchOld,
+      Runnable traverseNextEntry) {
+    this.project = gitRepository.getProject();
+    this.gitRepository = gitRepository;
+    this.graphTable = graphTable;
+    this.gitMacheteBranchOld = gitMacheteBranchOld;
+    this.traverseNextEntry = traverseNextEntry;
+  }
+
+  @UIEffect
+  public void execute() {
+    // we need to re-retrieve the gitMacheteBranch as its syncToRemote status could have changed after TraverseSyncToParent
+    val repositorySnapshot = graphTable.getGitMacheteRepositorySnapshot();
+    if (repositorySnapshot == null) {
+      LOG.warn("repositorySnapshot is null");
+      return;
+    }
+
+    val gitMacheteBranch =  repositorySnapshot.getManagedBranchByName(gitMacheteBranchOld.getName());
+    if (gitMacheteBranch == null) {
+      LOG.warn("gitMacheteBranch is null");
+      return;
+    }
+
+    val ongoingRepositoryOperationType = repositorySnapshot.getOngoingRepositoryOperation().getOperationType();
+    if (ongoingRepositoryOperationType != NO_OPERATION) {
```

Comment:
In particular it's for the case when someone closes the dialog for conflict resolution during the rebase:
<img width="387" alt="image" src="https://user-images.githubusercontent.com/19799111/207086391-69db53a5-9ebb-47ce-a504-05640da32413.png">

but actually we catch em all here! :surprised-pikachu:

Reply:
fyi: i ve tested the happy path here (with the conflicts) - works like gold 🥇